src/test/kotlin/org/dif/crypto/key/SenderKeySelectorTest.kt

```diff
@@ -0,0 +1,289 @@
+                senderKeySelector.findAnonCryptKeys(did)
+            }
+
+            assertEquals(expected, actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<DIDDocNotResolvedException> {
+                senderKeySelector.findAuthCryptKeys(JWM.ALICE_DID, did)
+            }
+
+            assertEquals(expected, actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<DIDDocNotResolvedException> {
+                senderKeySelector.findAuthCryptKeys(did, JWM.ALICE_DID)
+            }
+
+            assertEquals(expected, actual.message)
+        }
+    }
+
+    @Test
+    fun `Test empty DID Doc`() {
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), AliceSecretResolverMock())
+
+        run {
+            val actual = assertFailsWith<DIDDocException> {
+                senderKeySelector.findSigningKey(JWM.ELLIE_DID)
+            }
+
+            assertEquals("The DID Doc '${JWM.ELLIE_DID}' does not contain compatible 'authentication' verification methods", actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<DIDDocException> {
+                senderKeySelector.findAnonCryptKeys(JWM.ELLIE_DID)
+            }
+
+            assertEquals("The DID Doc '${JWM.ELLIE_DID}' does not contain compatible 'keyAgreement' verification methods", actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<DIDDocException> {
+                senderKeySelector.findAuthCryptKeys(JWM.ELLIE_DID, JWM.BOB_DID)
+            }
+
+            assertEquals("The DID Doc '${JWM.ELLIE_DID}' does not contain compatible 'keyAgreement' verification methods", actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<DIDDocException> {
+                senderKeySelector.findAuthCryptKeys(JWM.ALICE_DID, JWM.ELLIE_DID)
+            }
+
+            assertEquals("The DID Doc '${JWM.ELLIE_DID}' does not contain compatible 'keyAgreement' verification methods", actual.message)
+        }
+    }
+
+    @Test
+    fun `Test incompatible Crypto`() {
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), CharlieSecretResolverMock())
+        val bobDIDUrl = "did:example:bob#key-p256-1"
+        val charlieDIDUrl = "did:example:charlie#key-x25519-1"
+
+        run {
+            val actual = assertFailsWith<DIDDocException> {
+                senderKeySelector.findAuthCryptKeys(JWM.CHARLIE_DID, bobDIDUrl)
+            }
+
+            assertEquals("The DID Doc '${JWM.CHARLIE_DID}' does not contain compatible 'keyAgreement' verification methods", actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<IncompatibleCryptoException> {
+                senderKeySelector.findAuthCryptKeys(charlieDIDUrl, bobDIDUrl)
+            }
+
+            assertEquals("The recipient '$bobDIDUrl' curve is not compatible to 'X25519'", actual.message)
+        }
+    }
+}
```

Comment:
Please add a test where Alice and Bob DID DOCs contain both compatible and incompatible crypto.
Example:
- Alice DID DOC: key-type1, key-type2, key-type3
- Bob DID DOC: key-type3
- Expected: key-type-3 is chosen